src/test/java/org/opentripplanner/transit/raptor/moduletests/D01_TimeDependentAccessTest.java

```diff
@@ -0,0 +1,241 @@
+package org.opentripplanner.transit.raptor.moduletests;
+
+import static org.junit.Assert.assertEquals;
+import static org.opentripplanner.transit.raptor._data.api.PathUtils.pathsToString;
+import static org.opentripplanner.transit.raptor._data.transit.TestRoute.route;
+import static org.opentripplanner.transit.raptor._data.transit.TestTransfer.walk;
+import static org.opentripplanner.transit.raptor._data.transit.TestTripSchedule.schedule;
+import static org.opentripplanner.transit.raptor.api.transit.RaptorSlackProvider.defaultSlackProvider;
+import static org.opentripplanner.util.time.TimeUtils.hm2time;
+
+import java.time.Duration;
+import org.junit.Before;
+import org.junit.Test;
+import org.opentripplanner.routing.algorithm.raptor.transit.AccessEgress;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.transit.raptor.RaptorService;
+import org.opentripplanner.transit.raptor._data.RaptorTestConstants;
+import org.opentripplanner.transit.raptor._data.transit.TestTransitData;
+import org.opentripplanner.transit.raptor._data.transit.TestTripSchedule;
+import org.opentripplanner.transit.raptor.api.request.RaptorProfile;
+import org.opentripplanner.transit.raptor.api.request.RaptorRequestBuilder;
+import org.opentripplanner.transit.raptor.rangeraptor.configure.RaptorConfig;
+
+/**
+ * FEATURE UNDER TEST
+ *
+ * Raptor should take into account time restrictions on access. If the time restrictions require it,
+ * there should be a wait before boarding the trip so that the access is traversed while "open".
+ */
+public class D01_TimeDependentAccessTest implements RaptorTestConstants {
+
+    private final TestTransitData data = new TestTransitData();
+    private final RaptorRequestBuilder<TestTripSchedule> requestBuilder =
+            new RaptorRequestBuilder<>();
+    private final RaptorService<TestTripSchedule> raptorService = new RaptorService<>(
+            RaptorConfig.defaultConfigForTest()
+    );
+
+    @Before
+    public void setup() {
+        data.add(
+                route("R1", STOP_1, STOP_2, STOP_3, STOP_4, STOP_5)
+                        .withTimetable(
+                                schedule("0:10, 0:15, 0:20, 0:25, 0:30")
+                        )
+        );
+        data.add(
+                route("R2", STOP_1, STOP_2, STOP_3, STOP_4, STOP_5)
+                        .withTimetable(
+                                schedule("0:15, 0:20, 0:25, 0:30, 0:35")
+                        )
+        );
+        data.add(
+                route("R3", STOP_1, STOP_2, STOP_3, STOP_4, STOP_5)
+                        .withTimetable(
+                                schedule("0:20, 0:25, 0:30, 0:35, 0:40")
+                        )
+        );
+        data.add(
+                route("R4", STOP_1, STOP_2, STOP_3, STOP_4, STOP_5)
+                        .withTimetable(
+                                schedule("0:25, 0:30, 0:35, 0:40, 0:45")
+                        )
```

Comment:
For this test I would just use one route with 4 trips:
```
data.add(
        route("R1", STOP_1, STOP_2, STOP_3, STOP_4, STOP_5).withTimetable(
                schedule("0:10 0:15 0:20 0:25 0:30"),
                schedule("0:15 0:20 0:25 0:30 0:35"),
                schedule("0:20 0:25 0:30 0:35 0:40"),
                schedule("0:25 0:30 0:35 0:40 0:45")
        )
);
```
I  also removed the  `,`  in the times - it obstruct reading, but that is just style. If you like it  keep it.

Reply:
+1, the commas were removed.